tests/loggers/test_mlflow_logger.py

```diff
@@ -456,7 +453,8 @@ def test_mlflow_logging_works(tmp_path, device):
                       eval_interval=eval_interval,
                       device=device)
     trainer.fit()
-    test_mlflow_logger._flush()
+    # Allow async logging to finish.
+    time.sleep(3)
```

Comment:
this scares me a bit, because we want to make sure all messages are sent at the end of a training run.

Reply:
This is because the program hasn't exited before our assert clause, for actual training the eventual consistency is guaranteed (exit in the middle of training or end of training).

Reply:
so the cleanup runs using `atexit` or something?

Reply:
exactly

Reply:
it would be nice i think if mlflow exposed a way to manually flush instead of having to wait for `atexit` to run, then we could call that explicitly instead of inserting a sleep into our tests.

Reply:
It makes sense to me to have a `flush` method to force flushing. The reason for us not having it is the async logging logic is tied to MLflow store instead of client, so it won't be a trivial feature as it looks like.

Let me know if you want to block on this `flush` feature, but it won't be available until our next patch release, which probably happens in a few weeks. I would like to have async logging feature available on the Mosaic side soon so that we can do more testing to get confidence for turning async logging on by default.